### PR TITLE
[no ticket][risk=no] e2e test: Click Create a New Notebook button is not working

### PR DESCRIPTION
**Pull request by aweng98**
This is a test code workaround to deal with puppeteer flaky `.click` function. I don't know why it doesn't work sometimes.

Example of failed CircleCI [job](https://app.circleci.com/pipelines/github/all-of-us/workbench/16886/workflows/e386dbd4-969b-45c0-9fab-3c7401546553/jobs/188705/tests).

```
failure: [
  'TimeoutError: waiting for XPath `//*[@id="popup-root"]//*[@role="dialog" and contains(@class, "after-open")]` failed: timeout 30000ms exceeded\n' +
    '    at new WaitTask (/home/circleci/workbench/e2e/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:505:34)\n' +
    '    at DOMWorld.waitForXPath (/home/circleci/workbench/e2e/node_modules/puppeteer/lib/cjs/puppeteer/common/DOMWorld.js:441:26)\n' +
    '    at Frame.waitForXPath (/home/circleci/workbench/e2e/node_modules/puppeteer/lib/cjs/puppeteer/common/FrameManager.js:866:51)\n' +
    '    at Page.waitForXPath (/home/circleci/workbench/e2e/node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js:1288:33)\n' +
    '    at NewNotebookModal.waitUntilVisible (/home/circleci/workbench/e2e/app/container.ts:34:21)\n' +
    '    at NewNotebookModal.waitForLoad (/home/circleci/workbench/e2e/app/modal/base-modal.ts:27:16)\n' +
    '    at WorkspaceAnalysisPage.createNotebook (/home/circleci/workbench/e2e/app/page/workspace-analysis-page.ts:38:17)\n' +
    '    at Object.<anonymous> (/home/circleci/workbench/e2e/tests/notebook/notebook-create-python.spec.ts:52:22)'
]
```

### CODE REVIEW

File: e2e/app/element/base-element.ts [hunk 1]
```diff
@@ -167,6 +167,9 @@ export default class BaseElement {
         await element.hover();
         await this.page.waitForTimeout(500);
       }
+      if (previousX === undefined || previousY === undefined) {
+        throw new Error(`Failed to click element because the it is not visible in the viewport. (${this.getXpath()})`);
+      }
```
Comment (aweng98) [inline e2e/app/element/base-element.ts:172]: Click will not work if the element was not visible. Throws error to end the test here.

File: e2e/app/element/base-element.ts [hunk 2]
```diff
@@ -115,59 +123,68 @@ export default class BaseElement {
...
         }
-        await element.hover();
+        // Half second sleep before getting values of (x, y) again.
         await this.page.waitForTimeout(500);
       }
-      return element.click(options);
+      // End the test if previousX or previousY is still undefined after waiting.
+      if (previousX === undefined || previousY === undefined) {
+        throw new Error(`Failed to click element because it is not visible. xpath: ${this.getXpath()}`);
+      }
```
Comment (aweng98) [inline e2e/app/element/base-element.ts:186]: I didn't realize until now the values of previousX and previousY could still be `undefined` after the while loop has reached max wait time. Clicking on a non-visible element just doesn't work. Puppeteer doesn't throw any error from click. It makes test flaky.

File: e2e/app/element/base-element.ts [hunk 3]
```diff
@@ -1,4 +1,12 @@
-import { BoxModel, ClickOptions, ElementHandle, NavigationOptions, Page, WaitForSelectorOptions } from 'puppeteer';
+import {
+  BoxModel,
+  ClickOptions,
+  ElementHandle,
+  JSHandle,
+  NavigationOptions,
+  Page,
+  WaitForSelectorOptions
+} from 'puppeteer';
```
Comment (aweng98) [inline e2e/app/element/base-element.ts:9]: `eslint --fix` reformatted this.

File: e2e/app/modal/new-notebook-modal.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ export default class NewNotebookModal extends Modal {
 
   async isLoaded(): Promise<boolean> {
     await waitForText(this.page, modalTitle, { container: this });
+    await this.name().exists(30000);
```
Comment (aweng98) [inline e2e/app/modal/new-notebook-modal.ts:20]: To ensure modal is truly ready.

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**